File: csharp/bob/Bob.cs
using System;
using System.Linq;
public static class Bob
{
    public static string Hey(string statement)
    {
        var sure = "Sure.";
        var yell = "Whoa, chill out!";
        var ignore = "Fine. Be that way!";
        var other = "Whatever.";
        
        if(string.IsNullOrWhiteSpace(statement))
        {
            return ignore;
        }
        else if(IsYell(statement) && hasLetter(statement))
        {
            return yell;
        }
        else if(statement.Trim().EndsWith("?"))
        {
            return sure;
        }
        return other;
    }

    private static bool IsYell(string str) => str.Where(x=>Char.IsLetter(x)).All(x=>Char.IsUpper(x));
    private static bool hasLetter(string str) => str.Any(x=>Char.IsLetter(x));
}

File: go/bob/bob.go
package bob

import (
	"strings"
	"unicode"
)

const testVersion = 3

func Hay(input string) string {
	isAllUpper := false
	for _, ch := range input {
		if unicode.IsLetter(ch) {
			if unicode.IsUpper(ch) {
				isAllUpper = true
			} else {
				isAllUpper = false
				break
			}
		}
		continue
	}
	if isAllUpper {
		return "Whoa, chill out!"
	} else if input[len(input)-1] == '?' {
		return "Sure."
	} else if len(strings.TrimSpace(input)) == 0 {
		return "Fine. Be that way!"
	} else {
		return "Whatever."
	}
}
